src/ACEtk/interpretation/DEDX1/CRTP.hpp

```diff
@@ -0,0 +1,110 @@
+template<typename Derived>
+class CRTP{
+  std::reference_wrapper<const Table> table;
+
+  static constexpr auto cm = dimwits::centi(dimwits::meter);
+  static constexpr auto cc = cm*cm*cm;  
+  static constexpr auto mev = dimwits::mega(dimwits::electronVolt);
+  using DenT = decltype(1.0/cc);
+  using TempT = decltype(1.0*mev);
+
+protected:
+  auto numEnergies() const {return this->table.get().data.NXS( 4 );}
+  auto numDensities() const {return this->table.get().data.NXS( 5 );}
+  auto numTemperatures() const {return this->table.get().data.NXS( 6 );}
+  auto gridStart() const {return this->table.get().data.JXS( 2 );}    
+  
+public:
+  CRTP( const Table& table ) : table(table) {}
+
+  auto logEnergies() const {
+    const auto length = this->numEnergies();
```

Comment:
Please move these functions to a separate file. The style guide states that every function that is more than one line long needs to be in it's own file. The files should be in the `src/` directory and have the same name as the function.

Reply:
I will create a few files for some of the accessors in efforts to keep the class < 100 lines.